spec.html

```diff
@@ -8587,6 +8587,8 @@ <h1>SetFunctionName ( _F_, _name_ [ , _prefix_ ] )</h1>
           1. Let _description_ be _name_'s [[Description]] value.
           1. If _description_ is *undefined*, set _name_ to the empty String.
           1. Else, set _name_ to the string-concatenation of *"["*, _description_, and *"]"*.
+        1. If _F_ has an [[InitialName]] internal slot, then
+          1. Set F.[[InitialName]] to _name_.
```

Comment:
There's a type mismatch here. `_name_` is a String and `_F_.[[InitialName]]` is expected to be a sequence of Unicode code points. You'll have to do the conversion somewhere.

Reply:
why can't InitialName be a string everywhere?

Reply:
Hmm, it's kinda complicated. Below, we say

> ... the portion of the returned String that would be matched by |PropertyName| must be the value of _func_.[[InitialName]].

in an attempt to make use of what we've defined as "[the source text matched by](https://tc39.es/ecma262/#sec-algorithm-conventions-syntax-directed-operations)". But that operates on a source text (a sequence of Unicode code points) and assumes a parse has occurred (which also is only defined for a sequence of Unicode code points). So we're being a little sloppy in that usage. But since we're already using Strings inappropriately there, I think it's fine for now to assign a String here and check below that the value of `_F_.[[InitialName]]` is a String. The editors can clean that up later if we feel it needs it.